Magazine/Interfaces/IGenericRepository.cs

```diff
@@ -8,4 +8,5 @@ public interface IGenericRepository<T> where T : class
     void Add(T entity);
     void Update(T entity);
     void Delete(T entity);
+    Task<bool> SaveChangesAsync();
```

Comment:
you are moving the unit of work responsibility to repositories, that's not a good idea as it variolates SRP
better to just call save changes internally w/o exposing this